OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -65,6 +74,7 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 				return NextActivity;
 
 			// Try to move within range
+			var maxRange = MaxRange();
```

Comment:
There is a bit of code smell by allowing the min and max ranges to be passed into the ctor and then modifying the max range (what if the caller had some non-weapon reason for the limit?).  You could avoid this by passing the weapon itself in instead of the individual ranges.